### DISCUSSION TITLE
TProfile2D implementation when loading from ROOT?

### DISCUSSION THREAD

**Discussion by bblidaru** [scikit-hep/boost-histogram#749] [Q&A] [answered]
Hi! 
Following my initial question from [here](https://stackoverflow.com/questions/72535600/rebinning-a-th2-histogram-from-root-using-uproot-and-then-plot-it-with-matplotli), I tried importing a TProfile2D from ROOT and plot it with `hist.plot()`.
This failed, as it is [not yet implemented](https://github.com/scikit-hep/uproot4/blob/6fea96e3e61814cdeb21a3527c215f1911759d9a/src/uproot/behaviors/TProfile2D.py#L105). 
However Jim remembers writing it himself :)
I am trying to understand if using a TProfile works and/or if I need some magic tweak, or if there is some TProfile equivalent.
Thanks!

**Comment by henryiii** [comment, marked as answer]
I'm going to be working on boost-histogram again next week, April 29-Jun 10 were really busy for me! To me TProfile2D should just be a 2D histogram with WeightedMean storage?

**Reply by bblidaru** [reply]
Precisely! The way I see it, the TProfile2D is a more appealing variant of TH3 :)

**Reply by bblidaru** [reply]
Hello Henry!
A small bump here asking if there were any update maybe from your side :)? 
Unfortunately, I still haven't found a reliable way of working with uproot and plotting rebinned TProfile2D with hist.
